packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -574,23 +550,41 @@ export async function launch({
           }
         } else {
           // And now we actually process the queued actions down here, during
-          // END_BLOCK, but still reentrancy-protected
+          // END_BLOCK, but still reentrancy-protected.
+
+          provideInstallationPublisher();
 
-          // Process our begin, queued actions, and end.
-          await processAction(beginBlockAction); // BEGIN_BLOCK
+          // First we allow the timer to poll, which might push work onto the
+          // kernel run-queue, and gets the first cycle.
```

Comment:
This comment, in this commit ("disentangle launch-chain some more") and the next, is not accurate: timer work gets pushed onto the *end* of the run-queue, never getting the first cycle. Let's change it to:

> We update the timer device at the start of each block, which might push work onto the end of the kernel run-queue (if any timers were ready to wake), where it will be followed by actions triggered by the block's swingset transactions.

Then, in the commit where we switch to cycles, add:

> If the queue was empty, the timer work gets the first "cycle", and might run to completion before the block actions get their own cycles.